Python_Engine/Query/VirtualEnvironment.cs

```diff
@@ -44,7 +44,9 @@ public static bool VirtualEnvironmentExists(string envName, PythonVersion python
             bool directoryExists = Directory.Exists(VirtualEnvironmentDirectory(envName));
             bool executableExists = File.Exists(VirtualEnvironmentExecutable(envName));
             bool kernelExists = Directory.Exists(VirtualEnvironmentKernel(envName));
-            bool versionMatches = Version(VirtualEnvironmentExecutable(envName)) == pythonVersion;
+            bool versionMatches = false;
+            if (executableExists)
+                versionMatches = Version(VirtualEnvironmentExecutable(envName)) == pythonVersion;
```

Comment:
The output from `VirtualEnvironmentExecutable()` should be stored and reused as discussed given it's called twice in this method currently.